instrumentation/base/lib/opentelemetry/instrumentation/base.rb

```diff
@@ -293,11 +293,15 @@ def enabled?(config = nil)
 
       # Checks compatability of `library_name` against the `development_dependency` declared in  instrumentations gemspec
       def compatible_version?
-        instrumentation_spec = Gem.loaded_specs[instrumentation_gem_name] || Gem::Specification.find_by_name(instrumentation_gem_name)
-        library_spec = Gem.loaded_specs[library_name] || Gem::Specification.find_by_name(library_name)
-
-        dependency = instrumentation_spec&.development_dependencies&.find { |spec| spec.name == library_spec.name }
-        dependency&.requirement&.satisfied_by?(library_spec.version) == true
+        begin
+          instrumentation_spec = Gem.loaded_specs[instrumentation_gem_name] || Gem::Specification.find_by_name(instrumentation_gem_name)
+          library_spec = Gem.loaded_specs[library_name] || Gem::Specification.find_by_name(library_name)
+
+          dependency = instrumentation_spec.development_dependencies.find { |spec| spec.name == library_spec.name }
+          dependency&.requirement&.satisfied_by?(library_spec.version) == true
+        rescue Gem::MissingSpecError
+          return false
```

Comment:
A debug or warn log here couldn’t hurt?

Reply:
Sure thing.

Reply:
I chose to use `handle_error`. Seem reasonable?

Reply:
👍